src/commonMain/kotlin/com/github/quillraven/fleks/world.kt

```diff
@@ -177,6 +177,60 @@ class WorldConfiguration(@PublishedApi internal val world: World) {
             // instead of resizing every time a system gets added to the configuration
             world.systems = it.systems.toTypedArray()
         }
+
+        setUpAggregatedFamilyHooks()
+    }
+
+    /**
+     * Extend [Family.addHook] and [Family.removeHook] with
+     * other objects that needed to triggered by the hooks.
+     */
+    private fun setUpAggregatedFamilyHooks() {
+
+        // Validate systems against illegal interfaces.
+        world.systems.forEach { system ->
+            // FamilyOnAdd and FamilyOnRemove interfaces are only meant to be used by IteratingSystem.
+            if (system !is IteratingSystem) {
+
+                if (system is IteratingSystem.FamilyOnAdd)
+                    throw FleksWrongSystemInterfaceException(system::class, IteratingSystem.FamilyOnAdd::class)
+
+                if (system is IteratingSystem.FamilyOnRemove)
+                    throw FleksWrongSystemInterfaceException(system::class, IteratingSystem.FamilyOnRemove::class)
+            }
+        }
+
+        // Register family hooks for IteratingSystem.FamilyOnAdd containing systems.
+        world.systems
+            .mapNotNull { system -> if (system is IteratingSystem && system is IteratingSystem.FamilyOnAdd) system else null }
+            .groupBy { system -> system.family }
+            .forEach { entry ->
+                val (family, systems) = entry
+                val ownHook = family.addHook
+                family.addHook = if (ownHook != null) { entity ->
+                    ownHook(world, entity)
+                    systems.forEach { system -> system.onAddEntity(entity) }
+                }
+                else { entity ->
```

Comment:
else in same line as '}' bracket of if. The same formatting issue is below with FamilyOnRemove